runtime/queries/metricsview_aggregation.go

```diff
@@ -139,18 +139,28 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 
 	selectCols := make([]string, 0, len(q.Dimensions)+len(q.Measures))
 	groupCols := make([]string, 0, len(q.Dimensions))
+	unnestClauses := make([]string, 0)
 	args := []any{}
 
 	for _, d := range q.Dimensions {
 		// Handle regular dimensions
 		if d.TimeGrain == runtimev1.TimeGrain_TIME_GRAIN_UNSPECIFIED {
-			col, err := metricsViewDimensionToSafeColumn(mv, d.Name)
+			dim, err := findDimension(mv, d.Name)
 			if err != nil {
 				return "", nil, err
 			}
-
-			selectCols = append(selectCols, fmt.Sprintf("%s as %s", col, safeName(d.Name)))
-			groupCols = append(groupCols, col)
+			rawColName := getMetricsViewDimensionName(dim)
+			col := safeName(rawColName)
+			unnestColName := safeName(fmt.Sprintf("%s_%s", "unnested", rawColName))
```

Comment:
1. There might be another column named `unnested_...` already :) Consider using the `tempName` util func instead
2. This var is not used in the `else` clause below, so can be moved into the `if` clause